src/audio/module_adapter/module/waves.c

```diff
@@ -684,6 +659,22 @@ static int waves_codec_init(struct processing_module *mod)
 		setup_cfg->avail = true;
 	}
 
+	ret = MaxxEffect_GetMessageMaxSize(waves_codec->effect, &waves_codec->request_max_bytes,
+					      &waves_codec->response_max_bytes);
+
+	if (ret) {
+		comp_err(dev, "cadence_codec_init() MaxxEffect_GetMessageMaxSize returned %d", ret);
+		return -EINVAL;
```

Comment:
we should just return ret (otherwise we loose the original error context).

Reply:
We use our internal error number system in MaxxEffect_GetMessageMaxSize() function. So, we decide to print the internal error code here and then transfer to sof error number.

Reply:
ah, ok makes sense.